client/components/ContextMenu.jsx

```diff
@@ -0,0 +1,118 @@
+/** @jsx h */
+import {h, Component} from 'preact';
+import cls from 'classnames';
+import {store} from '../store';
+
+import s from './ContextMenu.css';
+
+export default class ContextMenu extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.boundingRect = this.node.getBoundingClientRect();
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.visible || nextProps.visible;
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.visible && !prevProps.visible) {
+      document.addEventListener('mousedown', this.handleDocumentMousedown);
+    } else if (prevProps.visible && !this.props.visible) {
+      document.removeEventListener('mousedown', this.handleDocumentMousedown);
+    }
+  }
+
+  render() {
+    const {visible} = this.props;
+    const containerClassName = cls({
+      [s.container]: true,
+      [s.hidden]: !visible
+    });
+    const itemClassName = cls({
+      [s.item]: true
+    });
+    return (
+      <ul ref={this.saveNode} className={containerClassName} style={this.getStyle()}>
+        <li className={itemClassName} onClick={this.handleClickHideChunk}>Hide chunk</li>
+        <li className={itemClassName} onClick={this.handleClickFilterToParents}>Show parent chunks</li>
+        <li className={itemClassName} onClick={this.handleClickFilterToChunk}>Hide all other chunks</li>
+        <hr/>
```

Comment:
I might have made the laziest possible interpretation of @th0r's request in https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/241#issuecomment-456334239 to add group separators here.

I could see either abstracting out the ContextMenu component so it can be passed a declarative config of options OR swapping out in favor of an existing open-source context menu, as there might be one that already has good UX thought put into it and that supports hierarchical menu options.